src/lib/layers/configurable-composite-layer.js

```diff
@@ -0,0 +1,91 @@
+import { CompositeLayer } from 'deck.gl';
+
+/* deck.gl util methods */
+function flatten(array, { filter = () => true, map = x => x, result = [] } = {}) {
+  // Wrap single object in array
+  if (!Array.isArray(array)) {
+    return filter(array) ? [map(array)] : [];
+  }
+  // Deep flatten and filter the array
+  return flattenArray(array, filter, map, result);
+}
+
+// Deep flattens an array. Helper to `flatten`, see its parameters
+function flattenArray(array, filter, map, result) {
+  let index = -1;
+  while (++index < array.length) {
+    const value = array[index];
+    if (Array.isArray(value)) {
+      flattenArray(value, filter, map, result);
+    } else if (filter(value)) {
+      result.push(map(value));
+    }
+  }
+  return result;
+}
+
+export default class ConfigurableCompositeLayer extends CompositeLayer {
+  _renderLayers() {
+    let { subLayers } = this.internalState;
+    if (subLayers && !this.needsUpdate()) {
+      // log.log(3, `Composite layer reused subLayers ${this}`, this.internalState.subLayers)();
+    } else {
+      subLayers = this.renderLayers();
+      subLayers = this.handleOverrides(subLayers);
```

Comment:
So this actually goes ahead and renders the initial layer, and then clones an override?

Again, I never considered that. Less performant but less intrusive in existing layers, and flexible.

Reply:
I believe when I wrote this, I was trying to keep the existing methods of `CompositeLayer` intact, thinking layer overrides would eventually go into `CompositeLayer` itself and that `renderLayers()` is overridden when creating a composite class. Also generally trying to reduce the surface area of the PR.